### ISSUE TITLE
Default keybind for "newline"

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/reedline#792]
Nushell pipelines are best split over multiple lines, and the `nufmt` [specification](https://github.com/nushell/nufmt/blob/main/docs/specification.md) recommends "wall of pipes" formatting.  E.g.:

```Nushell
ls
| where size > 0kb
| insert name_length {$in.name | str length}
```

However, that's extremely tedious to create "out of the box" with Nushell/Reedline at the moment, since a <kbd>Enter</kbd> only creates a newline in this case if there's a **trailing** pipe character on the existing line.  E.g.:

* `ls |` + <kbd>Enter</kbd> will create a newline which can be followed by the next expression, but
* `ls` + <kbd>Enter</kbd> executes the command (as it should - I'm not advocating for that to change ;-).

In order to create the "wall of pipes", you have to start with the first form `ls |`, hit <kbd>Enter</kbd> , then continue your `| pipeline`, and then (if you're pedantic) remove the trailing pipe from the previous line.  You could also do something like start the pipeline with a `(` open-paran to keep <kbd>Enter</kbd> from executing the pipeline until complete.

While it's relatively easy for a user to create a keybinding in their config to map <kbd>Alt</kbd>+<kbd>Enter</kbd> (assuming macOS <kbd>⌘</kbd>?), given the importance of newline in Nushell CLI-usage, I'd recommend this be a built-in keybinding.

Note:  This is built-in to the Fish shell.

**Comment by fdncred** [comment]
Ya, I've always thought the pipes belong on the right side for the same reasons. I have this in my keybindings because the problem with alt-enter is that it's not free in all terminals.
```
  {
    name: insert_newline
    modifier: alt
    keycode: char_i
    mode: [emacs vi_normal vi_insert]
    event: { edit: insertnewline }
  }
  {
    name: insert_newline
    modifier: alt
    keycode: enter
    mode: [emacs vi_normal vi_insert]
    event: { edit: insertnewline }
  }
```
But I wouldn't have a problem with adding these to the default_config.nu and in the rust code.

**Comment by sholderbach** [comment]
Closed by #854
